### PR TITLE
Bump LDAP from 1.26 to 2.3

### PR DESCRIPTION
**Pull request by basil**
Noticed this `TODO` in `war/pom.xml` that can easily be resolved. jenkinsci/ldap-plugin#49 has been present in upstream LDAP releases since 2.0, and the latest upstream LDAP release (2.3) depends only on Mailer 1.32.1, which we are also bundling as a detached plugin as of ~#5185~ #4938.

### CODE REVIEW

**Review by res0nance — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]
There iis no reason to keep old dependencies in the bundle. I am +1 for a bulk update

**Review by markjacksonfishing — APPROVED** [review]

**Comment by jglick** [comment]
This should not have been `internal` or `skip-changelog`: it forces users to update to 2.3 or later.

(FTR: appeared in 2.279, not in LTS.)

**Comment by MarkEWaite** [comment]
Proposed correction in https://github.com/jenkins-infra/jenkins.io/pull/4142

**Comment by basil** [comment]
> it forces users to update to 2.3 or later

My apologies; my intention in #5256, #5255, and #5244 wasn't to force users to do anything. I thought those PRs were merely updating already bundled dependencies to newer versions.

If those PRs have caused disruption, I would like to understand the nature of the disruption and the reason the change caused any disruption. We may want to possibly revert those changes. If we want to keep those changes, I can submit an update to the changelog to cover #5256 and #5255 as well.

**Comment by jglick** [comment]
https://github.com/jenkinsci/jenkins/blob/2ddf35821c119b7c219c2b0d7f113291f683f16c/core/src/main/java/hudson/PluginManager.java#L768-L773

**Comment by jglick** [comment]
Generally speaking, we only update detached plugins when we are forced to, for example because there was a security advisory. `ldap` is a bit of a special case because since https://github.com/jenkinsci/ldap-plugin/pull/48 in 1.26 it physically bundled Acegi Security and so it _is_ desirable to upgrade it to 2.x so that we are no longer including that old library in core. I would however suggest upgrading only to 2.0, as 2.1 introduced some other behavioral changes which have caused regressions that are still being fixed.

**Comment by basil** [comment]
Oh wow, that was definitely news to me. I now see how this would be an undesirable change; I'll submit a new PR to bump down LDAP back to 2.0 and to revert #5255 and #5256. My apologies again.

**Comment by jvz** [comment]
The LDAP upgrade beyond 2.0 will need the updates to mailer and email-ext that are in review right now.

**Pull request merged**
